qiskit/quantum_info/states/probabilitycache.py

```diff
@@ -0,0 +1,152 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+""" Probabilities Cache class. """
+
+from __future__ import annotations
+from qiskit.quantum_info.states.quantum_state import QuantumState
+
+
+class ProbabilityCache:
```

Comment:
it looks like possibly this could be used if this PR were to go the route of continuing to allow multiple target bitstrings, functionality could be added to get the most closely measured bitstring to what is stored in the dict here